back/app/services/reset_password_service.rb

```diff
@@ -1,34 +1,65 @@
+# frozen_string_literal: true
+
 require 'jwt'
 
+# Contains business logic for handling password resets for users via emails
+# like generating, validating tokens and sending emails.
 class ResetPasswordService
 
-  def generate_reset_password_token user
+  # Generates a password reset token for the specified user, valid for 1 hour.
+  # @param user [User] The user for whom the token should be generated
+  # @return [String] returns the generated token as a `String`
+  def generate_reset_password_token(user)
     payload = {
       id: user.id,
-      exp: (Time.now+1.hour).to_i
+      exp: 1.hour.from_now.to_i
     }
 
-    JWT.encode payload, secret, 'HS256'
+    JWT.encode(payload, secret, 'HS256')
   end
 
-  def token_valid? user, token
-    begin
-      payload = JWT.decode token, secret, true, { algorithm: 'HS256'}
-      payload[0]["id"] == user.id
-    rescue JWT::ExpiredSignature
-      return false
-    end
+  # Schedules an email to the user with instructions on how to reset their password
+  # after a token has been generated (and saved to the user record).
+  # @param user [User] the recipient of the user
+  # @param token [String] the password reset token for the user
+  # @return [void]
+  def send_email_later(user, token)
+    url = url_for(user, token)
+    ResetPasswordMailer.with(user: user, password_reset_url: url)
+                       .send_reset_password
+                       .deliver_later(priority: 1)
```

Comment:
Did you have a chance to test that priority is properly passed and used by Que? I wouldn't trust our Que-ActiveJob integration :smile:

Reply:
Good point, I'll add a unit test for it.

Reply:
So there is no easy way of doing it with the `have_enqueued_job` job matcher from RSpec, but I manually checked the queue and the priority is indeed set as expected.

Reply:
Great!